cybersemics/em
# Fix cursor position on paste

Fixes #2050.


Due to issues with async thunks in this action's tests, we're currently testing manually in the browser, see instructions below.

Additionally, this PR ~disables TypeScript output while running tests and~ adds a VSCode `launch.json` that enables testing and debugging within VSCode. In order to execute a test file, simply hit `F5` from the target test file. You may place breakpoints or `debugger` statements to pause execution.

### Test Setup

1. Create a new thought "a".
2. Depending on the test case, toggle `showHiddenThoughts`.
3. Paste one of the following snippets. Make sure to "Paste and Match Style" (Cmd+Shift+V).

#### Focus on first imported non-meta thought

Input:
```
- =note
  - Test
- b
- c
```

Expected cursor: `['a', 'b']`

#### Maintain focus when pasting only meta-attributes

Input:
```
- =note
  - Red
- =style
  - color
    - blue
```

Expected cursor: `['a']`

#### Focus on first imported thought when `showHiddenThoughts` is true

Enable `showHiddenThoughts`.

Input:
```
- =note
  - Red
- b
- c
```

Expected cursor: `['a', '=note']`

#### Focus on first imported thought when `showHiddenThoughts` is true and pasting only meta-attributes

Enable `showHiddenThoughts`.

Input:
```
- =note
  - Red
- =style
  - color
    - blue
```

Expected cursor: `['a', '=note']`
